jose/parse.go

```diff
@@ -358,6 +358,19 @@ func guessJWKAlgorithm(ctx *context, jwk *JSONWebKey) {
 	}
 }
 
+func guessOpaqueSigner(key crypto.PrivateKey) crypto.PrivateKey {
```

Comment:
Should this be called `guessSigner`, since it can also return other types of signers, specifically the `X25519Signer` (and will leave others untouched)?

Reply:
`X25519Signer` is an implementation of the `OpaqueSigner` interface.

Reply:
The method intends to return an OpaqueSigner only if necessary.